src/v/kafka/requests/request_context.h

```diff
@@ -67,46 +56,25 @@ struct request_header {
 
 std::ostream& operator<<(std::ostream&, const request_header&);
 
-class response;
-using response_ptr = ss::foreign_ptr<std::unique_ptr<response>>;
-
 class request_context {
 public:
     request_context(
-      ss::sharded<cluster::metadata_cache>& metadata_cache,
-      cluster::topics_frontend& topics_frontend,
+      ss::lw_shared_ptr<connection_context> conn,
       request_header&& header,
       iobuf&& request,
-      ss::lowres_clock::duration throttle_delay,
-      kafka::group_router_type& group_router,
-      cluster::shard_table& shard_table,
-      ss::sharded<cluster::partition_manager>& partition_manager,
-      ss::sharded<coordinator_ntp_mapper>& coordinator_mapper,
-      ss::sharded<fetch_session_cache>& fetch_session_cache) noexcept
-      : _metadata_cache(&metadata_cache)
-      , _topics_frontend(&topics_frontend)
+      ss::lowres_clock::duration throttle_delay) noexcept
+      : _conn(std::move(conn))
       , _header(std::move(header))
       , _reader(std::move(request))
-      , _throttle_delay(throttle_delay)
-      , _group_router(&group_router)
-      , _shard_table(&shard_table)
-      , _partition_manager(&partition_manager)
-      , _coordinator_mapper(&coordinator_mapper)
-      , _fetch_session_cache(&fetch_session_cache) {
+      , _throttle_delay(throttle_delay) {
         // XXX: don't forget to extend the move ctor
     }
     ~request_context() noexcept = default;
     request_context(request_context&& o) noexcept
-      : _metadata_cache(o._metadata_cache)
-      , _topics_frontend(o._topics_frontend)
+      : _conn(std::move(o._conn))
       , _header(std::move(o._header))
       , _reader(std::move(o._reader))
-      , _throttle_delay(o._throttle_delay)
-      , _group_router(o._group_router)
-      , _shard_table(o._shard_table)
-      , _partition_manager(o._partition_manager)
-      , _coordinator_mapper(o._coordinator_mapper)
-      , _fetch_session_cache(o._fetch_session_cache) {}
+      , _throttle_delay(o._throttle_delay) {}
     request_context& operator=(request_context&& o) noexcept {
         if (this != &o) {
             this->~request_context();
```

Comment:
Nobody would make it a virtual class, right?!

You could `static_assert` `is_same`, for good measure, and in case it's copy-pasted.

Reply:
yeh this is a weird workaround for some noexcept stuff in gcc-9